lib/testing.ex

```diff
@@ -0,0 +1,206 @@
+defmodule Uploadex.Testing do
+  @moduledoc """
+  This module simplifies testing and assertions involving `Uploadex`.
+
+  ## Usage in Tests
+
+  The most convenient way to use `Uploadex.Testing` is to `use` the module:
+
+      use Uploadex.Testing
+
+  That will define all the helper functions you'll need to make assertions.
+
+  ## Some examples of how to use the functions
+
+  ```elixir
+  # Asserting that 2 files were stored
+  assert_stored_files_count(2)
+
+  # Asserting the filenames of the stored files
+  assert_stored_files(["file1.pdf", "file2.pdf"])
+
+  # Asserting the filenames of the stored files ignoring the order
+  assert_stored_files(["file2.pdf", "file1.pdf"], ignore_order: true)
+
+  # Asserting that no file was stored
+  refute_stored_files()
+
+  # Asserting that 1 files was deleted
+  assert_deleted_files_count(1)
+
+  # Asserting the filenames of the deleted files
+  assert_deleted(["file3.pdf"])
+
+  # Asserting the filenames of the deleted files ignoring the order
+  assert_deleted_files(["file2.pdf", "file1.pdf"], ignore_order: true)
+
+  # Asserting that no file was deleted
+  refute_deleted_files()
+  ```
+
+  If you need a more low level API, you can directly use the `Uploadex.TestStorage` functions.
+  """
+
+  import ExUnit.Assertions, only: [assert: 2]
+
+  alias Uploadex.TestStorage
+
+  @doc """
+  Imports all functions from `Uploadex.Testing`
+  and defines a setup callback to start the `Uploadex.TestStorage`.
+  """
+  defmacro __using__(_opts) do
+    quote do
+      import Uploadex.Testing
+
+      setup :start_test_storage
+    end
+  end
+
+  @doc """
+  Starts the `Uploadex.TestStorage`.
+
+  You can use it as a `ExUnit` setup callback:
+
+      setup :start_test_storage
+
+  """
+  @spec start_test_storage(context :: map()) :: :ok
+  def start_test_storage(_ctx \\ %{}) do
+    TestStorage.start_link()
+    :ok
+  end
+
+  @doc """
+  Asserts that the given `expected_count` number of files was stored.
+  """
+  @spec assert_stored_files_count(expected_count :: integer) :: true
```

Comment:
shouldn't the signature be `:: boolean()` for these functions?

Reply:
Don't think so: https://github.com/elixir-lang/elixir/blob/main/lib/ex_unit/lib/ex_unit/assertions.ex#L393

At least for when the `assert` receives a message, it always returns true or raise

Reply:
Huh, good to know 🙂